src/ui/components/App.jsx

```diff
@@ -8,22 +8,24 @@ import { Main } from './Main/Main';
 import { PageNotFound } from './PageNotFound/PageNotFound';
 import { Error } from './Error/Error';
 import { routes } from '../../config/routes';
+import { ViewportProvider } from '../utils/viewport';
 
 export const App = () => (
-  <>
+  <ViewportProvider>
```

Comment:
Probably we can move it into index.jsx - we keep all providers like StoreProvider / ThemeProvider there